File: packages/cli/src/clipboard.ts
import { spawn } from 'node:child_process'

const COMMANDS: Record<string, [string, string[]][]> = {
  darwin: [['pbcopy', []]],
  win32: [['clip', []]],
  linux: [
    ['wl-copy', []],
    ['xclip', ['-selection', 'clipboard']],
    ['xsel', ['--clipboard', '--input']],
  ],
}

function tryCopy(cmd: string, args: string[], text: string): Promise<boolean> {
  return new Promise((resolve) => {
    const child = spawn(cmd, args, { stdio: ['pipe', 'ignore', 'ignore'] })
    child.on('error', () => resolve(false))
    child.on('close', (code) => resolve(code === 0))
    child.stdin.end(text)
  })
}

/** Best effort: returns false when no clipboard tool is available. */
export async function copyToClipboard(text: string): Promise<boolean> {
  for (const [cmd, args] of COMMANDS[process.platform] ?? []) {
    if (await tryCopy(cmd, args, text)) return true
  }
  return false
}

File: internal/clipboard/clipboard.go
// Package clipboard copies text with the platform's clipboard command.
package clipboard

import (
	"os/exec"
	"runtime"
	"strings"
)

var commands = map[string][][]string{
	"darwin":  {{"pbcopy"}},
	"windows": {{"clip"}},
	"linux": {
		{"wl-copy"},
		{"xclip", "-selection", "clipboard"},
		{"xsel", "--clipboard", "--input"},
	},
}

// Copy is best effort: it returns false when no clipboard tool is available.
func Copy(text string) bool {
	for _, argv := range commands[runtime.GOOS] {
		cmd := exec.Command(argv[0], argv[1:]...)
		cmd.Stdin = strings.NewReader(text)
		if cmd.Run() == nil {
			return true
		}
	}
	return false
}
